File: src/main/scala/br/com/fabioaraujo/calcula/produto/controller/HoraTrabalhadaController.scala
package br.com.fabioaraujo.calcula.produto.controller

import br.com.fabioaraujo.calcula.produto.dto.{HoraTrabalhada, Item}
import br.com.fabioaraujo.calcula.produto.service.{HoraTrabalhadaService, ItemService}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._

@RestController
@RequestMapping(path = Array("hora"))
class HoraTrabalhadaController(@Autowired val horaTrabalhadaService: HoraTrabalhadaService) {

  @PostMapping(path = Array("atualizar"))
  @ResponseBody
  def criar(@RequestBody horaTrabalhada : HoraTrabalhada): HoraTrabalhada = {
    val id = horaTrabalhadaService.criar(horaTrabalhada)
    horaTrabalhada
  }

  @GetMapping(path = Array("get"))
  @ResponseBody
  def getItem(): HoraTrabalhada = {
    horaTrabalhadaService.getHora()
  }

}

File: src/main/java/br/com/fabioaraujo/calcula/produto/controller/HoraTrabalhadaController.java
package br.com.fabioaraujo.calcula.produto.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import br.com.fabioaraujo.calcula.produto.dto.HoraTrabalhada;
import br.com.fabioaraujo.calcula.produto.service.HoraTrabalhadaService;

@RestController
@RequestMapping("hora-trabalhada")
public class HoraTrabalhadaController{
	@Autowired 
	private HoraTrabalhadaService horaTrabalhadaService;

  @PutMapping()
  @ResponseBody
  public ResponseEntity<Void> criarOuAtualizar(@RequestBody HoraTrabalhada horaTrabalhada){
	horaTrabalhada.setId(1L);
	horaTrabalhadaService.criar(horaTrabalhada);
    return ResponseEntity.noContent().build();
  }

  @GetMapping()
  @ResponseBody
  private HoraTrabalhada getItem() {
    return horaTrabalhadaService.getHora();
  }

}